packages/server/src/unstable-core-do-not-import/middleware.ts

```diff
@@ -102,14 +118,57 @@ export type MiddlewareFunction<
     meta: TMeta | undefined;
     signal: AbortSignal | undefined;
     next: {
+      /**
+       * Call the next middleware in the chain
+       * @see https://trpc.io/docs/server/middlewares
+       */
       (): Promise<MiddlewareResult<TContextOverridesIn>>;
-      <$ContextOverride>(opts: {
-        ctx?: $ContextOverride;
-        input?: unknown;
-      }): Promise<MiddlewareResult<$ContextOverride>>;
+      /**
+       * @internal
+       */
       (opts: {
-        getRawInput: GetRawInputFn;
+        [nextInputSymbol]: unknown;
       }): Promise<MiddlewareResult<TContextOverridesIn>>;
+      /**
+       * @deprecated
+       * You don't need to use the `ctx`-key on the `next()` anymore, simply call `next()` with an object to override the context
+       *
+       * **Before:**
+       * ```ts
+       * return next({ ctx: { foo: 'bar' } })
+       * ```
+       *
+       * **After:**
+       * ```ts
+       * return next({ foo: 'bar' })
+       * ```
+       * @see https://trpc.io/docs/server/middlewares#context-extension
+       */
+      <$ContextOverride>(
+        opts:
+          | {
+              ctx: $ContextOverride;
+            }
+          | {
+              /**
+               * @deprecated
+               * The ability to override the `input`-key will be removed in v12 - please open an issue if you need this feature
```

Comment:
will it be removed? isn't input overriding a valid tool in tRPC?

Reply:
I can't think of a usecase where it's needed

Less is more, reducing api surface area feels like a good thing